app/src/androidTest/java/com/github/se/di/module/TestModule.kt

```diff
@@ -0,0 +1,338 @@
+  fun provideFirebaseFirestore(): FirebaseFirestore {
+    return provideMockFirebasefirestore() // Provide the mock instance for testing
+  }
+
+  private fun provideMockFirebasefirestore(): FirebaseFirestore {
+    val globalMockGeopoint = mock(GeoPoint::class.java)
+    `when`(globalMockGeopoint.latitude).thenReturn(46.51827)
+    `when`(globalMockGeopoint.longitude).thenReturn(6.619265)
+    // mock firebasefirestore for porfileRepositoryFirestore
+    var myProfile =
+        Profile(
+            "12345",
+            "IceBreakrr end-to-end",
+            Gender.MEN,
+            Timestamp.now(),
+            "",
+            "",
+            emptyList(),
+            null,
+            null)
+    val profilesDocumentSnapshots =
+        listOf(
+            mock(DocumentSnapshot::class.java),
+            mock(DocumentSnapshot::class.java),
+            mock(DocumentSnapshot::class.java),
+            mock(DocumentSnapshot::class.java),
+            mock(DocumentSnapshot::class.java),
+            mock(DocumentSnapshot::class.java),
+            mock(DocumentSnapshot::class.java),
+            mock(DocumentSnapshot::class.java),
+            mock(DocumentSnapshot::class.java),
+            mock(DocumentSnapshot::class.java),
+            mock(DocumentSnapshot::class.java),
+            mock(DocumentSnapshot::class.java))
+    val profilesDocumentReference =
+        listOf(
+            mock(DocumentReference::class.java),
+            mock(DocumentReference::class.java),
+            mock(DocumentReference::class.java),
+            mock(DocumentReference::class.java),
+            mock(DocumentReference::class.java),
+            mock(DocumentReference::class.java),
+            mock(DocumentReference::class.java),
+            mock(DocumentReference::class.java),
+            mock(DocumentReference::class.java),
+            mock(DocumentReference::class.java),
+            mock(DocumentReference::class.java),
+            mock(DocumentReference::class.java))
+    val profilesTaskDocumentSnapshot =
+        listOf(
+            mock(Task::class.java) as Task<DocumentSnapshot>,
+            mock(Task::class.java) as Task<DocumentSnapshot>,
+            mock(Task::class.java) as Task<DocumentSnapshot>,
+            mock(Task::class.java) as Task<DocumentSnapshot>,
+            mock(Task::class.java) as Task<DocumentSnapshot>,
+            mock(Task::class.java) as Task<DocumentSnapshot>,
+            mock(Task::class.java) as Task<DocumentSnapshot>,
+            mock(Task::class.java) as Task<DocumentSnapshot>,
+            mock(Task::class.java) as Task<DocumentSnapshot>,
+            mock(Task::class.java) as Task<DocumentSnapshot>,
+            mock(Task::class.java) as Task<DocumentSnapshot>,
+            mock(Task::class.java) as Task<DocumentSnapshot>)
+    val profiles = Profile.Companion.getMockedProfiles()
+    // set the profilesDocumentSnapshot to return the mock profiles informations :
+    for (i in 0 until 12) {
```

Comment:
_nitpick_ : I've put 12 in a private variable in the object. 
_problem_ : until in kotlin is exclusive, so I iterate through 12 elements but I will still put the syntax you shared because it is more concise

Reply:
Oh my days you are right, sorry I got mixed up the 0..10 notation is fully inclusive and yours was working. Sorry for the bad advice (but at least now you know another cool syntax)

Reply:
Yep thanks! ^^